src/plugins/intel_cpu/src/nodes/executors/shl/shl_eltwise.cpp

```diff
@@ -53,6 +43,26 @@ bool ShlEltwiseExecutorBuilder::isSupported(const EltwiseAttrs& eltwiseAttrs,
         return false;
     }
 
+    // check whether input and output layouts are equal
+    if(srcDescs.front()->hasLayoutType(LayoutType::nCsp16c) || srcDescs.front()->hasLayoutType(LayoutType::nCsp8c)) {
+        DEBUG_LOG("ShlEltwise does not support 'nCsp16c' or 'nCsp8c' layouts");
+        return false;
+    }
+    const auto unifiedLayout = srcDescs.front()->hasLayoutType(LayoutType::ncsp) ? LayoutType::ncsp : LayoutType::nspc;
+    const auto unifiedRank = srcDescs.front()->getShape().getRank();
+    auto has_unified_layout = [unifiedLayout, unifiedRank](const MemoryDescPtr& desc) {
+        if (desc->hasLayoutType(LayoutType::nspc)) {    // ensure the same rank
+            if (desc->getShape().getRank() != unifiedRank)
```

Comment:
Since now `ShlExecutor` works with `BlockedDims` (L107, L110), I think we need to compare ranks of `BlockedDims`, not of planar shapes. Because `BlockedDims` and `Shape` may have different ranks and this check can be incorrect.
What do you think?

Reply:
Make sense! It promises coherent behavior inside `ShlExecutor`!